libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java

```diff
@@ -50,20 +55,119 @@
+        XContentType.YAML,
+        () -> randomRealisticUnicodeOfCodepointLength(3145730)
+    );
+
+    public void testStringOffLimit() throws IOException {
+        final XContentType xContentType = randomFrom(XContentType.values());
+
+        final String field = randomAlphaOfLengthBetween(1, 5);
+        final String value = OFF_LIMIT_GENERATORS.get(xContentType).get();
+
+        try (XContentBuilder builder = XContentBuilder.builder(xContentType.xContent())) {
+            builder.startObject();
+            if (randomBoolean()) {
+                builder.field(field, value);
+            } else {
+                builder.field(field).value(value);
+            }
+            builder.endObject();
+
+            try (XContentParser parser = createParser(xContentType.xContent(), BytesReference.bytes(builder))) {
+                assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+                assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
+                assertEquals(field, parser.currentName());
+                assertEquals(XContentParser.Token.VALUE_STRING, parser.nextToken());
+                if (xContentType != XContentType.YAML) {
+                    assertThrows(StreamConstraintsException.class, () -> parser.text());
+                } else {
+                    assertThrows(JacksonYAMLParseException.class, () -> parser.nextToken());
+                }
+            }
+        }
+    }
+
+    public void testString() throws IOException {
+        final XContentType xContentType = randomFrom(XContentType.values());
+
+        final String field = randomAlphaOfLengthBetween(1, 5);
+        final String value = GENERATORS.get(xContentType).get();
+
+        try (XContentBuilder builder = XContentBuilder.builder(xContentType.xContent())) {
+            builder.startObject();
+            if (randomBoolean()) {
+                builder.field(field, value);
+            } else {
+                builder.field(field).value(value);
+            }
+            builder.endObject();
+
+            final String text;
+            try (XContentParser parser = createParser(xContentType.xContent(), BytesReference.bytes(builder))) {
+                assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+                assertEquals(XContentParser.Token.FIELD_NAME, parser.nextToken());
+                assertEquals(field, parser.currentName());
+                assertEquals(XContentParser.Token.VALUE_STRING, parser.nextToken());
+
+                text = parser.text();
+
+                assertEquals(XContentParser.Token.END_OBJECT, parser.nextToken());
+                assertNull(parser.nextToken());
+            }
+
+            assertThat(text, hasLength(value.length()));
+
+            switch (xContentType) {
+                case CBOR:
+                case SMILE:
+                    assertThat(text, instanceOf(String.class));
+                    break;
```

Comment:
Why not fall through here?

Reply:
Copy / paste from other test, will fix that, thanks!